src/oas.md

```diff
@@ -44,6 +44,31 @@ Each template expression in the path MUST correspond to a path parameter that is
 
 The value for these path parameters MUST NOT contain any unescaped "generic syntax" characters described by [RFC3986](https://tools.ietf.org/html/rfc3986#section-3): forward slashes (`/`), question marks (`?`), or hashes (`#`).
 
+The path templating expression is defined by the following [ABNF](https://tools.ietf.org/html/rfc5234) syntax
+
+```abnf
+; OpenAPI Path Templating ABNF syntax
+path-template                  = path [ query-marker query ]
+path                           = slash *( path-segment slash ) [ path-segment ]
+path-segment                   = 1*( path-literal / template-expression )
+query                          = *( query-literal )
+query-literal                  = 1*( unreserved / pct-encoded / sub-delims / ":" / "@" / "/" / "?" / "&" / "=" )
+query-marker                   = "?"
+slash                          = "/"
+path-literal                   = 1*( unreserved / pct-encoded / sub-delims / ":" / "@" )
+template-expression            = "{" template-expression-param-name "}"
+template-expression-param-name = 1*( unreserved / pct-encoded / sub-delims / ":" / "@" )
```

Comment:
The only explicit restriction we state is that

> Each template expression in the path MUST correspond to a path parameter

and there are no restrictions on path parameter names.

The only implicit restriction is that the `}` cannot be used in a template expression because we don't define an escaping mechanism for it.

Reply:
so you'd extend what's in there in essence? I double checked RFC5234 and there does not appear to be any way so simply say "anything but this character", and I think it's on purpose.

What do you think is missing?  `%[]^"'<>?` ?

Reply:
Trick question: which ABNF rule is equivalent to regexp `^[^}]+$` 😎

Reply:
that's effectively what I was looking for, couldn't find an equivalent, but maybe I missed something?

Reply:
We'd probably have to list all characters except the `}`, which raises the question what "all characters" are:
* all octets `%x00-FF`, or
* just all characters allowed in URIs: `unreserved / reserved / pct-encoded`

One could argue that a path template is not restricted to URI characters and is just a string that becomes a URI path _after_ replacing all template expressions with their values.

Reply:
could we start with a more restrictive set (unreserved and al) and maybe open it up based on the feedback we get in another patch release?

Reply:
based on @handrews 's suggestion, should we go with that instead?

`%x00-79 / %x7C / %x7E-FF ; every unicode character except { and }`

I'm worried this will be too broad as it includes special characters like CR/LF/NUL space etc...

Reply:
We could omit the nonprinting control characters and start with `%x20`, or even exclude the space and start with `%x21`.

Reply:
There are unicode characters beyond %xFF.

Reply:
Then we should probably also exclude 7F to A0 as well. Anything else?
https://symbl.cc/en/unicode-table/

Reply:
> There are unicode characters beyond %xFF.

@karenetheridge do we think it's broadly needed? could we stick with basic latin and latin1 (minus what we're planning to exclude already) and see whether people ask us to extend it later in a patch release?

Reply:
@baywet we should not forbid anything that is currently allowed.  So unless the spec says you cannot use those characters, then you an use them and they should be allowed.  Keep in mind that [Appendix C.4.4](https://spec.openapis.org/oas/v3.1.1#illegal-variable-names-as-parameter-names) shows an example of a parameter with the heart emoji as a name.

We can probably get away with excluding the non-printing control characters, as that's pretty pathological.  I'm not sure we should forbid the space character.

Reply:
The most generous rule would be
```suggestion
template-expression-param-name = 1*( %x00-7C / %x7E-10FFFF ) ; every unicode character except "}"
```
This relies on path templates being represented as JSON or YAML strings and thus using UTF-8.

Reply:
@ralfhandl should we also at least exclude `{` ? it'd probably make parser's lives much easier.

Reply:
@handrews?

Reply:
Thanks for all this additional information!

I think we're reaching a consensus to this

```anbf
%x00-79 / %x7C / %x7E-10FFFF ; every UTF8 character except { and }
```

Thoughts?

Reply:
> Trick question: which ABNF rule is equivalent to regexp `^[^}]+$` 😎

[SABNF](https://github.com/ldthomas/apg-js2/blob/master/SABNF.md) does allow that, be we want to keep to standard ABNF which does not